src/org/github/jamm/MemoryLayoutSpecification.java

```diff
@@ -70,7 +70,10 @@ public static MemoryLayoutSpecification getEffectiveMemoryLayoutSpecification()
             } else {
 
                 // In other cases, it's a 64-bit uncompressed OOPs object model
-                objectHeaderSize = 16; // mark word (8 bytes) + class word (8 bytes)
+                // Prior to Java 15, the use of compressed class pointers assumed the use of compressed oops.
+                // This was changed in Java 15 by JDK-8241825 (https://bugs.openjdk.org/browse/JDK-8241825).
+                objectHeaderSize = VM.useCompressedClassPointers() ? 12  // mark word (8 bytes) + class word (4 bytes)
```

Comment:
Unfortunately, the only way to do proper testing is through the JVM flag otherwise you end up with different results for the  Spec strategy and the other strategy. I do not see why those flags are system dependent. They should work the same for each systems.

Reply:
What do you want to test exactly?

Reply:
With Java 17 and `-Xmx64g`:
`Jamm starting with: java.version='17.0.6', java.vendor='Oracle Corporation', instrumentation=true, unsafe=true, Memory Layout: [objectHeaderSize=12 , arrayHeaderSize=16, objectAlignment=8, referenceSize=8, contendedPaddingWidth=128]`
With Java 17 and` -Xmx64g -XX:-UseCompressedClassPointers`:
`Jamm starting with: java.version='17.0.6', java.vendor='Oracle Corporation', instrumentation=true, unsafe=true, Memory Layout: [objectHeaderSize=16 , arrayHeaderSize=24, objectAlignment=8, referenceSize=8, contendedPaddingWidth=128]`
as you can see the option change the value of the `objectHeaderSize`